frontend/packages/dev-console/src/components/pipelines/modals/common/PipelineSecretSection.tsx

```diff
@@ -1,66 +1,50 @@
 import * as React from 'react';
-import { Formik, useField, useFormikContext, FormikValues } from 'formik';
+import { useFormikContext, FormikValues } from 'formik';
 import { PlusCircleIcon } from '@patternfly/react-icons';
 import { Button } from '@patternfly/react-core';
 import { ExpandCollapse } from '@console/internal/components/utils';
-import { SecretType } from '@console/internal/components/secrets/create-secret';
 import { SecretModel } from '@console/internal/models';
-import { k8sCreate } from '@console/internal/module/k8s';
-import {
-  associateServiceAccountToSecret,
-  getSecretAnnotations,
-} from '../../../../utils/pipeline-utils';
-import { SecretAnnotationId } from '../../const';
-import { advancedSectionValidationSchema } from './validation-utils';
+import { SecretKind } from '@console/internal/module/k8s';
+import { getSecretAnnotations } from '../../../../utils/pipeline-utils';
 import SecretForm from './SecretForm';
 import SecretsList from './SecretsList';
+import { secretFormDefaultValues } from './utils';
 
 import './PipelineSecretSection.scss';
 
-const initialValues = {
-  secretName: '',
-  annotations: { key: SecretAnnotationId.Image, value: '' },
-  type: SecretType.dockerconfigjson,
-  formData: {},
-};
-
 type PipelineSecretSectionProps = {
   namespace: string;
 };
 
 const PipelineSecretSection: React.FC<PipelineSecretSectionProps> = ({ namespace }) => {
-  const [secretOpenField] = useField<boolean>('secretOpen');
-  const { setFieldValue } = useFormikContext<FormikValues>();
+  const {
+    setFieldValue,
+    values: { newSecrets, secretForm },
+    setFieldTouched,
+  } = useFormikContext<FormikValues>();
 
-  const handleSubmit = (values, actions) => {
-    actions.setSubmitting(true);
+  const handleSubmit = () => {
+    setFieldValue(`secretForm`, secretFormDefaultValues);
+    setFieldTouched(`secretForm`, false);
     const newSecret = {
       apiVersion: SecretModel.apiVersion,
       kind: SecretModel.kind,
       metadata: {
-        name: values.secretName,
+        name: secretForm.secretName,
         namespace,
-        annotations: getSecretAnnotations(values.annotations),
+        annotations: getSecretAnnotations(secretForm.annotations),
       },
-      type: values.type,
-      stringData: values.formData,
+      type: secretForm.type,
+      stringData: secretForm.formData,
     };
-    k8sCreate(SecretModel, newSecret)
-      .then((resp) => {
-        actions.setSubmitting(false);
-        setFieldValue(secretOpenField.name, false);
-        associateServiceAccountToSecret(resp, namespace);
-      })
-      .catch((err) => {
-        actions.setSubmitting(false);
-        setFieldValue(secretOpenField.name, false);
-        actions.setStatus({ submitError: err.message });
-      });
+    const secrets: SecretKind[] = newSecrets;
+    secrets.push(newSecret as SecretKind);
+    setFieldValue(`newSecrets`, secrets);
```

Comment:
Before submitting, you're probably going to want to make sure your names don't collide. I am not testing this right now but if you try to add two secrets of the same name, I think you allow it and will error out on the form submit.

Reply:
@andrewballantyne should we check this on secret form submission? do we need to notify the user if there is a name collision or simply omit that secret object ?

Reply:
Before submit of the secret form, it should do it... otherwise you'll get into a situation where you cannot fix the problem. Maybe we want to allow edit of temporary ones? 🤔

Reply:
right now, I am handling this during the modal form submission. any error related to the creation of secret resource if occurs I am preventing the submission of the modal form and I'm displaying the error to the user.

Reply:
If we want to allow the user to edit  pending secrets can we do it in another story ?

Reply:
Let's do it in another story. We'll run it through UX.

Reply:
Today, prevent the duplication of name. I think that's all you need to worry about for now.